helper/resource/testing_new.go

```diff
@@ -106,8 +106,17 @@ func runNewTest(t testing.T, c TestCase, helper *tftest.Helper) {
 
 		if step.ImportState {
 			err := testStepNewImportState(t, c, helper, wd, step, appliedCfg)
-			if err != nil {
-				t.Fatal(err)
+			if step.ExpectError != nil {
+				if err == nil {
+					t.Fatalf("Step %d/%d error running import: expected an error but got none", i+1, len(c.Steps))
+				}
+				if !step.ExpectError.MatchString(err.Error()) {
+					t.Fatalf("Step %d/%d error running import, expected an error with pattern, no match on: %s", i+1, len(c.Steps), err)
```

Comment:
Would it be possible to return the `ExpectError` regular expression in the error message for easier troubleshooting?

```suggestion
					t.Fatalf("Step %d/%d error running import, expected an error with pattern (%s), no match on: %s", i+1, len(c.Steps), step.ExpectError.String(), err)
```

Reply:
definitely would be helpful 👍 in V1 the error msg used `"Step %d, expected error:\n\n%s\n\nTo match:\n\n%s\n\n",
						i, err, step.ExpectError))` , happy to commit something similar or the one suggested